src/modules/peek/Peek.FilePreviewer/Previewers/UnsupportedFilePreviewer/UnsupportedFilePreviewer.cs

```diff
@@ -96,17 +96,13 @@ public Task<bool> LoadIconPreviewAsync()
         {
             return TaskExtension.RunSafe(async () =>
             {
-                if (CancellationToken.IsCancellationRequested)
-                {
-                    _cancellationTokenSource = new CancellationTokenSource();
-                    return;
-                }
+                CancellationToken.ThrowIfCancellationRequested();
 
                 // TODO: Get icon with transparency
                 IconHelper.GetIcon(Path.GetFullPath(File.Path), out IntPtr hbitmap);
                 await Dispatcher.RunOnUiThread(async () =>
                 {
-                    var iconBitmap = await GetBitmapFromHBitmapAsync(hbitmap);
+                    var iconBitmap = await GetBitmapFromHBitmapAsync(hbitmap, CancellationToken);
```

Comment:
I would maybe add some CancellationToken.ThrowIfCancellationRequested calls before we set IconPreview, and in ImagePreview whenever we're about to set the Preview property in LoadFullQualityImageAsync, LoadHighQualityThumbnailAsync, and LoadLowQualityThumbnailAsync